pkg/backup/backup_compaction_test.go

```diff
@@ -270,6 +271,48 @@ crdb_internal.json_to_pb(
 			t, db, []string{"foo"}, "'nodelocal://1/backup'", opts,
 		)
 	})
+
+	t.Run("pause resume and cancel", func(t *testing.T) {
+		db.Exec(t, "CREATE TABLE foo (a INT, b INT)")
+		defer func() {
+			db.Exec(t, "DROP TABLE foo, bar")
+		}()
+		db.Exec(t, "INSERT INTO foo VALUES (1, 1)")
+		start := getTime()
+		db.Exec(t, fmt.Sprintf(fullBackupAostCmd, start))
+		db.Exec(t, "INSERT INTO foo VALUES (2, 2)")
+		db.Exec(t, incBackupCmd)
+		db.Exec(t, "CREATE TABLE bar (a INT, b INT)")
+		db.Exec(t, "INSERT INTO bar VALUES (1, 1)")
+		db.Exec(t, incBackupCmd)
+		db.Exec(t, "INSERT INTO bar VALUES (2, 2)")
+		db.Exec(t, incBackupCmd)
+		db.Exec(t, "INSERT INTO bar VALUES (3, 3)")
+		end := getTime()
+		db.Exec(t, fmt.Sprintf(incBackupAostCmd, end))
+		db.Exec(t, "SET CLUSTER SETTING jobs.debug.pausepoints = 'backup.after.details_has_checkpoint'")
+		defer func() {
+			db.Exec(t, "SET CLUSTER SETTING jobs.debug.pausepoints = ''")
+		}()
+
+		jobID := startCompaction(start, end)
+		jobutils.WaitForJobToPause(t, db, jobID)
+		db.Exec(t, "RESUME JOB $1", jobID)
+		waitForSuccessfulJob(t, tc, jobID)
+		validateCompactedBackupForTables(t, db, []string{"foo", "bar"}, "'nodelocal://1/backup'")
```

Comment:
I've realized in my testing that this lets through a couple of false negatives. `validateCompactedBackupForTables` simply restores from the latest backup. If the tests were created in isolation, this might not be too big of an issue, but since they all share the same cluster, one test that may have failed to create a compaction would actually just restore from a previous test.

I think we can fix this by having `validateCompactedBackupForTables` take in the `start` and `end` of the compaction, and it should also validate that a backup spanning that timespan exists. But I'll do this in another PR.

Reply:
could you instead use a different collection dir for each subtest instead of `nodelocal://1/backup` for everything?

Reply:
Oh yeah that would do the trick. I do think having an extra check that makes sure that there exists a backup that spans the entire time from `start` to `end` is good though.

Reply:
I'll do this in a separate PR though